src/main/java/org/spongepowered/api/Server.java

```diff
@@ -86,6 +89,21 @@
     Optional<Player> getPlayer(String name);
 
     /**
+     * Gets a {@link Player} by their name, as seen by the
+     * provided viewer.
+     *
+     * <p>This only works for online players.</p>
+     *
+     * <p><b>Note: Do not use names for persistent storage, the
+     * Notch of today may not be the Notch of yesterday.</b></p>
+     *
+     * @param name The name to get the player from
+     * @param viewer The viewer of the request
+     * @return {@link Player} or Optional.empty() if not found
+     */
+    Optional<Player> getPlayer(String name, @Nullable ContextViewer viewer);
```

Comment:
I think the fact that this is on `Server` and is named `getPlayer` is missleading. Maybe this method (and similar) should be on `ContextViewer`, since it's the context viewer that has the properties (what player can it see etc)